### ISSUE TITLE
After user clicks "Create service instance", there's no feedback about success or failure

### ISSUE THREAD

**Issue by brittag** [18F/cg-dashboard#457]
Currently when you go to an org's Marketplace and fill out the "Create service instance" form, you click the final "Create service instance" button, and then the form disappears. You don't get feedback on whether you successfully created the instance, and you don't get instructions for what to do next.

Let's add helpful feedback here, so that users know that it worked (and what to do next). After successful creation, let's provide a note saying something like this:
- **name** service instance is now available in **name** space, which is in your **name** organization. You can bind that service instance to an app [using the command line](https://docs.cloud.gov/apps/managed-services/).

Or after an unsuccessful effort at creation, something like this:
- Something went wrong while creating this service instance. Let us know at cloud-gov-support@gsa.gov (or #cloud-gov-support if you're in 18F), and we'll figure out what happened. You can also [try using the command line to create a service instance](https://docs.cloud.gov/apps/managed-services/).

Some notes from Slack:
 - Feeling hesitant to tackle before we work on marketplace page, but maybe I'll check out current usage in GA
 - From analytics, opening the “create service instance” form has been done by 10 users 17 times total since Sept 1st. (as of ~12/2)
 - So, it happens to be the most popular event we track, but not by a lot

Notes from subsequent discussion:
 - Do we have patterns for this? Bound route indicator when it fails
 - Editing app limits is a major change
 - UI doesn’t have a good way to indicate that this is happening
 - Would be good to say whether or not the action is in progress/has completed, maybe display the text success
 - Indicator that action has happened
 - Services panel so you can see service instance has been added

Implementation notes:
 - We'll plan on an MVP version to fix this for now, replacing "create service instance" button on success or failure with an appropriate message/dialog ("your service instance has been created" or error state with opportunity to try again or cancel) (@thisisdano will paste some screens here)
 - Could follow model for loading messages

Acceptance criteria:
 - [x] When clicking "Create service instance," appropriate messaging/dialog is displayed

**Comment by thisisdano** [comment]
This is a component of https://github.com/18F/cg-dashboard/issues/763

**Comment by suprenant** [comment]
Discussed in our grooming session just now. Essentially, we're not going to pick this up right away but will keep our eyes on it as we move through the current sprint:

Do we have patterns for this?
Bound route indicator when it fails

Editing app limits
Major change
UI doesn’t have a good way to indicate that this is happening
Usability issue – don’t understand what the UI is doing
Feels really broken, especially in the limits/editing case
Would be good to say whether or not the action is in progress/has completed, maybe display the text success
Will be more polished 
On old UI, won’t have app usage and limits
A lot will be more clear when Marketplace is in space
Indicator that action has happened
Will also be services panel so you can see service instance has been added

**Comment by suprenant** [comment]
Thinking of waiting on this until this page is further developed.

**Comment by suprenant** [comment]
@thisisdano @vz3: @msecret and I just discussed this. Given that we won't be picking up the Marketplace page in this PI, what are your thoughts of designing an MVP success/failure state that could be included in the current page here?

![screen shot 2017-01-03 at 4 51 39 pm](https://cloud.githubusercontent.com/assets/1452369/21626545/5f803752-d1d7-11e6-9988-c60f8ae7ecfb.png)

**Comment by thisisdano** [comment]
From the marketplace prototype:

- - -

**Button default state**

![marketplace-plansimpleok 2x](https://cloud.githubusercontent.com/assets/11464021/21657843/ecadc24c-d278-11e6-88b5-b2d38f90a6dc.png)

- - -

**Buttons replaced with loading indicator**

![marketplace-plansimplecreating 2x](https://cloud.githubusercontent.com/assets/11464021/21657851/f5731bc0-d278-11e6-8629-2f4eb24eaae1.png)

- - -

**Success message appears. In success case, the action panel could be removed after ~2sec.**

![marketplace-plansimplecreated 2x](https://cloud.githubusercontent.com/assets/11464021/21657855/fb0eab3a-d278-11e6-8f96-41427b19d31c.png)

- - -

**Or an error message, with an opportunity to try again, or to cancel. (Error message should be more verbose, descriptive than this "friendly" one.**

![marketplace-plansimplenotcreated 2x](https://cloud.githubusercontent.com/assets/11464021/21657861/0030a938-d279-11e6-8bff-6efcd749fe18.png)

**Comment by adborden** [comment]
This was completed by https://github.com/18F/cg-dashboard/pull/944

**Comment by adborden** [comment]
:+1: confirmation appears after success.

**Comment by brittag** [comment]
I just tried this on staging, and this confirmation of success seems to disappear after only a couple of seconds. Could it stick around until it gets dismissed?

Also the created message gets a few extra capital letters due to styling (they're not in [the code here](https://github.com/18F/cg-dashboard/blob/master/static_src/components/create_service_instance.jsx):

![Uploading Screen Shot 2017-01-27 at 5.24.49 PM.png…]()

**Comment by adborden** [comment]
That suggestion sounds good to me. I agree it kind of flashes with limited time to read and comprehend it. I'd put that in a new issue though. I think there are a few issues hanging around regarding messaging for success/failure and forms.

**Comment by brittag** [comment]
Thanks, filed a new issue here: https://github.com/18F/cg-dashboard/issues/991
